lib/beaker/dsl/helpers.rb

```diff
@@ -1340,6 +1340,38 @@ def fact(name, opts = {})
         fact_on(default, name, opts)
       end
 
+      # Snapshot provided host
+      #
+      # Only vagrant hosts are supported
+      # and you must have https://github.com/scalefactory/vagrant-multiprovider-snap installed
+      #
+      # @param host [Host] The host to snapshot
+      # @param name [String] The name to use as the snapshot name
+      def take_snapshot_on(host, name)
+        host.do_take_snapshot(name)
```

Comment:
Instead of sending this to the host, let's send to the hypervisor.  I think that we should split out a separate chunk of the DSL to be suts.rb or boxes.rb or nodes.rb that refers to actions that can be applied to hosts through their respective hypervisors.  Then we'll also need to make the hypervisor hash controlled by beaker available to those methods.  Then the call through system will work.